### ISSUE TITLE
How to control data samples in a batch

### ISSUE THREAD

**Issue by taokz** [OFA-Sys/OFA#303]
Hi! 

Thank you for your amazing work. I have a question about how to control the number of data samples in a batch. According to your paper, you said that 

We mix all the pretraining data within each batch, which contains 2, 048 vision&language samples, 256 object detection samples, 256 image-only samples and 512 text-only samples.

However, I didn't find where to control the data samples in the code. Could you provide some hints? 

Besides, could you explain the iterations in the logs? For example, when fine-tuning a caption dataset (~4000 training samples and ~1000 validation samples) with a batch size of 8. But the log shows that there are only 130 instead of 500 iterations during training. So the fine-tuning works on the validation dataset rather than the training dataset?

I would appreciate your help. Thanks.

**Comment by logicwong** [comment]
@taokz 

1. For controlling the data samples, see [snippet1](https://github.com/OFA-Sys/OFA/blob/5fd17e780d1c8074b177a927ec68068c95690ac1/data/pretrain_data/unify_dataset.py#L468-L476), [snippet2](https://github.com/OFA-Sys/OFA/blob/5fd17e780d1c8074b177a927ec68068c95690ac1/data/pretrain_data/unify_dataset.py#L617-L636), and [snippet3](https://github.com/OFA-Sys/OFA/blob/5fd17e780d1c8074b177a927ec68068c95690ac1/criterions/label_smoothed_cross_entropy.py#L176-L194)

2. We set `update_freq=4`, `update_freq` means **gradient accumulation**. The effective batch size of single GPU is equal to 32 (batch_size * update_freq)
